environments/local/local_test.go

```diff
@@ -225,14 +232,125 @@ func TestSixNode(t *testing.T) {
 	err = localEnv.StartNetwork()
 	require.NoError(t, err)
 
-	time.Sleep(30 * time.Second) // todo change this to a polling approach
-	for _, node := range networkCfg.Nodes {
-		c := client.NewClient("http://" + node.GetAPIAddr())
-		peers, err := c.GetPeers()
+	defer func() {
+		err = localEnv.StopNetwork()
+		require.NoError(t, err)
+	}()
+
+	pollingWhileConnectingPeers(t, networkCfg.Nodes, 5)
+}
+
+func TestSixNodesGalactica(t *testing.T) {
+	var sixNodesGalacticaNetwork *network.Network
+	require.NotPanics(t, func() { sixNodesGalacticaNetwork = preset.LocalSixNodesGalacticaNetwork() })
+
+	localEnv := NewLocalEnv()
+	_, err := localEnv.LoadConfig(sixNodesGalacticaNetwork)
+	require.NoError(t, err)
+
+	err = localEnv.StartNetwork()
+	require.NoError(t, err)
+
+	defer func() {
+		err = localEnv.StopNetwork()
 		require.NoError(t, err)
+	}()
 
-		require.GreaterOrEqual(t, len(peers), 0)
+	clients := pollingWhileConnectingPeers(t, sixNodesGalacticaNetwork.Nodes, 5)
+
+	deployAndAssertShanghaiContract(t, clients[0], preset.SixNNAccount1)
+}
+
+func pollingWhileConnectingPeers(t *testing.T, nodes []node.Node, expectedPeersLen int) []*client.Client {
+	// Polling approach with timeout
+	timeout := time.After(1 * time.Minute)
+	tick := time.Tick(5 * time.Second)
+
+	clients := make([]*client.Client, 0)
+outer:
+	for {
+		select {
+		case <-timeout:
+			t.Fatal("timed out waiting for nodes to connect")
+		case <-tick:
+			allConnected := true
+			for _, node := range nodes {
+				c := client.NewClient("http://" + node.GetAPIAddr())
+				peers, err := c.GetPeers()
+				require.NoError(t, err)
+				if len(peers) != expectedPeersLen {
+					allConnected = false
+					clients = clients[:0]
+					break
+				}
+				clients = append(clients, c)
+			}
+			if allConnected {
+				break outer
```

Comment:
How about `return clients` instead of using the `outer` label?

Reply:
Done, thanks